src/main/java/com/epam/architecture/SOAPws/handlers/UserHandler.java

```diff
@@ -0,0 +1,42 @@
+package com.epam.architecture.SOAPws.handlers;
+
+import com.epam.architecture.roles.AuthorizationUtil;
+import com.epam.architecture.roles.RoleEnum;
+import com.epam.architecture.userinterface.LibraryService;
+import jakarta.xml.soap.SOAPException;
+import jakarta.xml.ws.handler.MessageContext;
+import jakarta.xml.ws.handler.soap.SOAPHandler;
+import jakarta.xml.ws.handler.soap.SOAPMessageContext;
+
+import javax.xml.namespace.QName;
+import java.util.Collections;
+import java.util.Set;
+
+public class UserHandler implements SOAPHandler<SOAPMessageContext> {
+    @Override
+    public Set<QName> getHeaders() {
+        return Collections.emptySet();
+    }
+
+    @Override
+    public boolean handleMessage(SOAPMessageContext soapMessageContext) {
+        if (!(boolean) soapMessageContext.get(MessageContext.MESSAGE_OUTBOUND_PROPERTY)) {
+            try {
+                return AuthorizationUtil.isAuthorizedRequest(soapMessageContext, RoleEnum.USER);
```

Comment:
Admin also have user rights 
```suggestion
                return AuthorizationUtil.isAuthorizedRequest(soapMessageContext, setOf(RoleEnum.USER,RoleEnum.Admin));
```